test/hotspot/jtreg/runtime/cds/appcds/test-classes/CustomAppLoader.java

```diff
@@ -40,7 +40,7 @@
     // args[2...] = arguments for the main class
     public static void main(String args[]) throws Throwable {
         File f = new File(args[0]);
-        URL[] classLoaderUrls = new URL[] {new URL("file://" + f.getCanonicalPath())};
+        URL[] classLoaderUrls = new URL[] {f.toURI().toURL()};
```

Comment:
Not sure if the above can handle relative path.
It is better to set f as follows:
`File f = Path.of(args[0]).toRealPath().toFile();`
Before the above, there should be a check if the length of args is 1.